core/src/toga/statusicons.py

```diff
@@ -0,0 +1,247 @@
+            bar or system tray.
+        :param text: A text label for the status icon. Defaults to the formal name of
+            the app.
+        :param on_press: The handler to invoke when the status icon is pressed.
+        """
+        super().__init__(icon=icon)
+        self.on_press = on_press
+
+        self._id = f"statusicon-{_py_id(self)}" if id is None else id
+        self._text = text if text is not None else toga.App.app.formal_name
+
+    def __repr__(self):
+        return f"<StatusIcon {self.text!r}: {self.id}>"
+
+    @property
+    def id(self) -> str:
+        """A unique identifier for the icon."""
+        return self._id
+
+    @property
+    def text(self) -> str:
+        """A text label for the icon."""
+        return self._text
+
+    @property
+    def on_press(self) -> toga.widgets.button.OnPressHandler:
+        """The handler to invoke when the status icon is pressed."""
+        return self._on_press
+
+    @on_press.setter
+    def on_press(self, handler: toga.widgets.button.OnPressHandler) -> None:
+        self._on_press = wrapped_handler(self, handler)
+
+
+class MenuStatusIcon(BaseStatusIcon, Group):
+    def __init__(
+        self,
+        id: str | None = None,
+        icon: IconContentT | None = None,
+        text: str | None = None,
+    ):
+        """
+        An item in a status bar or system tray that displays a menu when pressed.
+
+        A ``MenuStatusIcon`` can be used as a :class:`~toga.Group` when defining
+        :class:`toga.Command` instances.
+
+        :param id: An identifier for the status icon.
+        :param icon: The icon, or icon resource, that will be displayed in the status
+            bar or system tray.
+        :param text: A text label for the status icon. Defaults to the formal name of
+            the app.
+        """
+        super().__init__(
+            id=f"menustatusitem-{_py_id(self)}" if id is None else id,
+            icon=icon,
+            text=(text if text is not None else toga.App.app.formal_name),
+        )
+
+    def __repr__(self):
+        return f"<MenuStatusIcon {self.text!r}: {self.id}>"
+
+
+class StatusIconSet(Sequence[BaseStatusIcon], Mapping[str, BaseStatusIcon]):
+    def __init__(self):
+        """An ordered collection of status icons.
+
+        The items in the set can be retrieved by instance, or by ID. When iterated, the
+        items are returned in the order they were added.
+        """
+        self.factory = get_platform_factory()
+        self._impl = self.factory.StatusIconSet(interface=self)
+
+        self.elements: dict[str, BaseStatusIcon] = {}
+        self.commands = CommandSet()
+
+    @property
+    def menu_status_icons(self):
+        """An iterator over the menu status icons that have been registered."""
+        return (icon for icon in self if isinstance(icon, MenuStatusIcon))
+
+    @property
+    def primary_menu_status_icon(self):
+        """The first menu status icon that has been registered.
```

Comment:
Do these need to be public? I can't think of why an app would use them.

Reply:
I can't think of a particular use case for them; but I couldn't think of a reason to *not* include them either. However, you're probably right that it's safer to hide them initially and make them public when/if there's a use case.